pkg/controllers/user/helm/common/common.go

```diff
@@ -162,10 +137,16 @@ func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
 		setValues = append(setValues, "--values", valuesYaml)
 	}
 	// `--set` values will overridden the user-supplied values.yaml file
-	if app.Spec.Answers != nil {
+	if app.Spec.Answers != nil || extraArgs != nil {
 		answers := app.Spec.Answers
 		var values = []string{}
 		for k, v := range answers {
+			if _, ok := extraArgs[k]; ok {
+				continue
+			}
+			values = append(values, fmt.Sprintf("%s=%s", k, v))
```

Comment:
do we need to sort the values?

Reply:
I don't think we need. Helm takes all the `--set` values as input. The sequence fo the `--set` values won't change the result of `helm install`.